controllers/cluster-registration/cluster_registration_controller.go

```diff
@@ -182,6 +196,71 @@ func (r *RegisteredClusterReconciler) Reconcile(computeContextOri context.Contex
 	return ctrl.Result{}, nil
 }
 
+func (r *RegisteredClusterReconciler) checkSynctargetExists(locationContext context.Context, regCluster *singaporev1alpha1.RegisteredCluster) (int, error) {
+
+	locationClusterName, _ := logicalcluster.ClusterFromContext(locationContext)
+	labels := RegisteredClusterNamelabel + "=" + regCluster.Name + "," + RegisteredClusterNamespacelabel + "=" + regCluster.Namespace + "," + RegisteredClusterWorkspace + "=" + strings.ReplaceAll(locationClusterName.String(), ":", "-")
+
+	syncTargetList, err := r.ComputeDynamicClient.Resource(clusterGVR).List(locationContext, metav1.ListOptions{
+		LabelSelector: labels,
+	})
+	if err != nil {
+		return 0, giterrors.WithStack(err)
+	}
+
+	r.Log.V(4).Info("Number of synctarget found with lables",
+		"number", len(syncTargetList.Items),
+		RegisteredClusterNamelabel, regCluster.Name,
+		RegisteredClusterNamespacelabel, regCluster.Namespace)
+
+	if len(syncTargetList.Items) == 1 {
+		return len(syncTargetList.Items), nil
+	}
+
+	return len(syncTargetList.Items), nil
+
+}
+
+func (r *RegisteredClusterReconciler) syncSyncTarget(computeContext context.Context, regCluster *singaporev1alpha1.RegisteredCluster, managedCluster *clusterapiv1.ManagedCluster) error {
+
+	logger := r.Log.WithName("syncSyncTarget").WithValues("namespace", regCluster.Namespace, "name", regCluster.Name, "managed cluster name", managedCluster.Name)
+
+	if status, ok := helpers.GetConditionStatus(regCluster.Status.Conditions, clusterapiv1.ManagedClusterConditionJoined); ok && status == metav1.ConditionTrue {
+
+		locationContext := logicalcluster.WithCluster(computeContext, logicalcluster.New(regCluster.Spec.Location))
+		locationClusterName, _ := logicalcluster.ClusterFromContext(locationContext)
+
+		syncTargetLength, err := r.checkSynctargetExists(locationContext, regCluster)
+		if err != nil {
+			return giterrors.WithStack(err)
+		}
+		if syncTargetLength == 0 {
+			syncTarget := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": workloadv1alpha1.SchemeGroupVersion.String(),
+					"kind":       "SyncTarget",
+					"metadata": map[string]interface{}{
+						"generateName": "registered-cluster-",
+						"labels": map[string]string{
+							RegisteredClusterNamelabel:      regCluster.Name,
+							RegisteredClusterNamespacelabel: regCluster.Namespace,
+							RegisteredClusterWorkspace:      strings.ReplaceAll(locationClusterName.String(), ":", "-"),
```

Comment:
RegisteredClusterWorkspace and ClusterNameAnnotation are the same string currently. ClusterNameAnnotation uses req.ClusterName for its value. Is that the same as what you calculate here?

Reply:
As noted above, I think you mean to use the RegisteredCluster's workspace, not the location workspace.

Reply:
Yeah, it should be workspace where registeredcluster is created.